File: services/typescript/koa/src/routes/router.ts
import Koa from "koa";
import Router from "@koa/router";
import type { State } from "app";
import {
  postHandler,
  getHandler,
  putHandler,
  patchHandler,
  deleteHandler
} from "./handlers/index.js";
import { invalidMethodMiddleware } from "./middleware/invalid_method.js";

export function registerRoutes(app: Koa, state: State): void {
  const router = new Router();

  router.post("/", postHandler(state));
  router.all("/", invalidMethodMiddleware);

  router.get("/:id", getHandler(state));
  router.put("/:id", putHandler(state));
  router.patch("/:id", patchHandler(state));
  router.delete("/:id", deleteHandler(state));
  router.all("/:id", invalidMethodMiddleware);

  app.use(router.routes());
  app.use(router.allowedMethods());
}

File: services/python/fastapi/src/routes/router.py
from fastapi import FastAPI

from routes.handlers import (
    post_handler,
    get_handler,
    delete_handler,
    put_handler,
    patch_handler,
)


def register_routes(app: FastAPI):
    app.post("/", status_code=201)(post_handler)
    app.get("/{id}", status_code=200)(get_handler)
    app.delete("/{id}", status_code=204)(delete_handler)
    app.put("/{id}", status_code=200)(put_handler)
    app.patch("/{id}", status_code=200)(patch_handler)
